dev/tests/static/testsuite/Magento/Test/Integrity/Magento/Framework/Cache/ConfigTest.php

```diff
@@ -0,0 +1,85 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types = 1);
+
+namespace Magento\Test\Integrity\Magento\Framework\Cache;
+
+/**
+ * Unit test of the cache configuration
+ */
+class ConfigTest extends \PHPUnit\Framework\TestCase
+{
+    /**
+     * Path to xsd schema file
+     * @var string
+     */
+    private $xsdSchema;
+
+    /**
+     * @var \Magento\Framework\Config\Dom\UrnResolver
+     */
+    private $urnResolver;
+
+    /**
+     * @var \Magento\Framework\TestFramework\Unit\Utility\XsdValidator
+     */
+    private $xsdValidator;
+
+    /**
+     * Setup environment for test
+     */
+    protected function setUp(): void
+    {
+        if (!function_exists('libxml_set_external_entity_loader')) {
+            $this->markTestSkipped('Skipped on HHVM. Will be fixed in MAGETWO-45033');
+        }
+        $this->urnResolver = new \Magento\Framework\Config\Dom\UrnResolver();
+        $this->xsdSchema = $this->urnResolver->getRealPath(
+            'urn:magento:framework:Cache/etc/cache.xsd'
+        );
+        $this->xsdValidator = new \Magento\Framework\TestFramework\Unit\Utility\XsdValidator();
+    }
+
+    /**
+     * Tests invalid configurations
+     *
+     * @param string $xmlString
+     * @param array $expectedError
+     * @dataProvider schemaCorrectlyIdentifiesInvalidXmlDataProvider
+     */
+    public function testSchemaCorrectlyIdentifiesInvalidXml(
+        string $xmlString,
+        array $expectedError
+    ): void {
+        $actualError = $this->xsdValidator->validate(
+            $this->xsdSchema,
+            $xmlString
+        );
+        $this->assertEquals($expectedError, $actualError);
+    }
+
+    /**
+     * Tests valid configurations
+     */
+    public function testSchemaCorrectlyIdentifiesValidXml(): void
+    {
+        $xmlString = file_get_contents(__DIR__ . '/_files/valid_cache_config.xml');
+        $actualResult = $this->xsdValidator->validate(
+            $this->xsdSchema,
+            $xmlString
+        );
+
+        $this->assertEmpty($actualResult);
+    }
+
+    /**
+     * Data provider with invalid xml array according to cache.xsd
+     */
+    public function schemaCorrectlyIdentifiesInvalidXmlDataProvider(): array
+    {
+        return include __DIR__ . '/_files/invalidCacheConfigXmlArray.php';
```

Comment:
Interesting approach to provide code samples.

Reply:
Hello @lbajsarowicz 
Thank you for your remark.
Magento already has the same data providers. Please take a look at the screenshot
<img width="967" alt="interesting-data-providers" src="https://user-images.githubusercontent.com/10788564/76943677-69913400-6908-11ea-91a3-88206cdfc685.png">
Thank you in advance.